### PR TITLE
#436 Configure GraphQL Query Limit

### PR DESCRIPTION
**Pull request by Clonk-HTML5**

### CODE REVIEW

**Comment by stefan-kamsker** [comment]
Hi,

thanks, we will check and merge it next week

happy new year to you!

Best regards!

**Comment by franzwilding** [comment]
Hey! Implementation looks not bad, however I think this should be a configuration option and not a service parameter. Please have a look at the implementation of maximum_nesting_level (https://github.com/unite-cms/unite-cms/blob/master/src/Bundle/CoreBundle/DependencyInjection/Configuration.php).

I also think, that this option should be called "maximum_query_limit" and not only "query_limit"

**Comment by Clonk-HTML5** [comment]
Hi,

i changed it!

File: src/Bundle/CoreBundle/Resources/config/services.graphql.yml [hunk 1]
```diff
@@ -1,76 +1,76 @@
-# Contains all graphql services
```
Comment (franzwilding) [inline src/Bundle/CoreBundle/Resources/config/services.graphql.yml]: There are many unnecessary git changes in this file. Maybe whitespace issues?

File: src/Bundle/CoreBundle/DependencyInjection/UniteCMSCoreExtension.php [hunk 2]
```diff
@@ -31,5 +31,9 @@ public function load(array $configs, ContainerBuilder $container)
         // Set maximum nesting level as first argument to schema type manager and maximum nesting type object type.
         $container->getDefinition('unite.cms.graphql.schema_type_manager')->setArgument(0, $config['maximum_nesting_level']);
         $container->getDefinition('UniteCMS\CoreBundle\SchemaType\Types\MaximumNestingLevelType')->setArgument(0, $config['maximum_nesting_level']);
+
+        // Set maximum query limit as fifth argument to schema type manager and maximum nesting type object type.
+        $container->getDefinition('UniteCMS\CoreBundle\SchemaType\Types\QueryType')->setArgument(5, $config['maximum_query_limit']);
```
Comment (franzwilding) [inline src/Bundle/CoreBundle/DependencyInjection/UniteCMSCoreExtension.php]: When replacing a service argument here, there should be a default argument (e.g. 100) in the services.yaml definition of the service as a fallback.

**Pull request merged**